### PR TITLE
fix: handle json files inside node_modules

### PR DESCRIPTION
**Pull request by igorzi**

### CODE REVIEW

File: src/plugin_deno_loader.ts [hunk 1]
```diff
@@ -380,7 +380,12 @@ export function denoLoaderPlugin(
         args: esbuild.OnLoadArgs,
       ): Promise<esbuild.OnLoadResult | null> {
         if (args.namespace === "file" && args.pluginData === IN_NODE_MODULES) {
-          const contents = await Deno.readFile(args.path);
+          if (args.path.endsWith(".json")) {
+            const contents = await Deno.readTextFile(args.path);
+            return { loader: "json", contents };
+          }
+          const path = args.path.endsWith(".js") ? args.path : `${args.path}.js`;
+          const contents = await Deno.readFile(path);
```
Comment (lucacasonato) [inline src/plugin_deno_loader.ts:388]: Is this needed?

**Review by lucacasonato — COMMENTED** [review]
This looks good. Can you add a test for this?

File: src/plugin_deno_loader.ts (hunk 1 above)
Comment (igorzi) [reply src/plugin_deno_loader.ts:388]: Without this I get errors like these:
```
✘ [ERROR] No such file or directory (os error 2): readfile '/Users/igorzi/Library/Caches/deno/deno_esbuild/object-inspect@1.13.1/node_modules/object-inspect/util.inspect' [plugin deno-loader]

    ../../../Users/igorzi/Library/Caches/deno/deno_esbuild/object-inspect@1.13.1/node_modules/object-inspect/index.js:68:26:
      68 │ var utilInspect = require('./util.inspect');
         ╵                           ~~~~~~~~~~~~~~~~

✘ [ERROR] No such file or directory (os error 2): readfile '/Users/igorzi/Library/Caches/deno/deno_esbuild/iconv-lite@0.4.24/node_modules/iconv-lite/lib/extend-node' [plugin deno-loader]

    ../../../Users/igorzi/Library/Caches/deno/deno_esbuild/iconv-lite@0.4.24/node_modules/iconv-lite/lib/index.js:148:12:
      148 │     require("./extend-node")(iconv);
          ╵             ~~~~~~~~~~~~~~~

✘ [ERROR] No such file or directory (os error 2): readfile '/Users/igorzi/Library/Caches/deno/deno_esbuild/iconv-lite@0.4.24/node_modules/iconv-lite/lib/streams' [plugin deno-loader]

    ../../../Users/igorzi/Library/Caches/deno/deno_esbuild/iconv-lite@0.4.24/node_modules/iconv-lite/lib/index.js:144:16:
      144 │         require("./streams")(iconv);
```

**Comment by igorzi** [comment]
more comprehensive fix in https://github.com/lucacasonato/esbuild_deno_loader/pull/118

**Pull request closed without merging**